Telemetry/CorePing.swift

```diff
@@ -66,10 +67,14 @@ class CorePing: TelemetryPing {
             "osversion": versionString,
             "device": "Apple-" + model,
             "arch": "arm",
-            "profileDate": profileDate,
-            "defaultSearch": profile.searchEngines.defaultEngine.id
+            "profileDate": profileDate
         ]
 
+        if defaultEngine.isCustomEngine {
+            out["defaultSearch"] = JSON.null
+        } else {
+            out["defaultSearch"] = defaultEngine.engineId
+        }
```

Comment:
...and move this block before the `out` dict above, creating a variable for the result like we do for the other fields.